manager/orchestrator/task.go

```diff
@@ -67,7 +67,28 @@ func IsTaskDirty(s *api.Service, t *api.Task) bool {
 		return false
 	}
 
-	return !reflect.DeepEqual(s.Spec.Task, t.Spec) ||
+	// Make shallow copies for the comparison.
+	specA, specB := s.Spec.Task, t.Spec
```

Comment:
nit: I would suggest renaming these to specService, specTask or something along those lines to make it more suggestive.